list.c

```diff
@@ -0,0 +1,345 @@
+                 int c10, int c11, int c12, int c13, int c14,
+                 int c15, int c16, int c17, int c18, int c19) {
+
+    assignString(string_buffer, c0, c1, c2, c3, c4, c5, c6, c7, c8, c9,c10, c11, c12, c13, c14, c15, c16, c17, c18, c19);
+    
+    print(string_buffer);
+}
+
+void print(int *s) {
+    while (*s != 0) {
+        putchar(*s);
+        s = s + 1;
+    }
+}
+
+int stringLength(int *s) {
+    int i;
+    
+    i = 0;
+    
+    while (*(s+i) != 0)
+        i = i + 1;
+    
+    return i;
+}
+
+void reverseString(int *s) {
+    int tmp;
+    int i;
+    int j;
+    
+    i = 0;
+    j = stringLength(s) - 1;
+    
+    while (i < j) {
+        tmp = *(s+i);
+        *(s+i) = *(s+j);
+        *(s+j) = tmp;
+        i = i + 1;
+        j = j - 1;
+    }
+}
+
+int* itoa(int n, int *s, int b, int a) {
+    // assert: b in {2,4,8,10,16}
+    
+    int i;
+    int sign;
+    
+    i = 0;
+    
+    sign = 0;
+    
+    if (n == 0) {
+        *s = '0';
+        
+        i = 1;
+    } else if (n < 0) {
+        sign = 1;
+        
+        if (b == 10) {
+            if (n == INT_MIN) {
+                *s = '8'; // rightmost decimal digit of 32-bit INT_MIN
+                
+                n = -(n / 10);
+                i = i + 1;
+            } else
+                n = -n;
+        } else {
+            if (n == INT_MIN) {
+                *s = '0'; // rightmost non-decimal digit of INT_MIN
+                
+                n = (rightShift(INT_MIN, 1) / b) * 2;
+                i = i + 1;
+            } else
+                n = rightShift(leftShift(n, 1), 1);
+        }
+    }
+    
+    while (n != 0) {
+        *(s+i) = n % b;
+        
+        if (*(s+i) > 9)
+            *(s+i) = *(s+i) - 10 + 'A';
+        else
+            *(s+i) = *(s+i) + '0';
+        
+        n = n / b;
+        i = i + 1;
+        
+        if (i == 1) {
+            if (sign) {
+                if (b != 10)
+                    n = n + (rightShift(INT_MIN, 1) / b) * 2;
+            }
+        }
+    }
+    
+    if (b != 10) {
+        while (i < a) {
+            *(s+i) = '0'; // align with zeros
+            
+            i = i + 1;
+        }
+    } else if (sign) {
+        *(s+i) = '-';
+        
+        i = i + 1;
+    }
+    
+    *(s+i) = 0; // null terminated string
+    
+    reverseString(s);
+    
+    return s;
+}
+
+int twoToThePowerOf(int p) {
+    // assert: 0 <= p < 31
+    return *(power_of_two_table + p);
+}
+
+int leftShift(int n, int b) {
+    // assert: b >= 0;
+    
+    if (b > 30)
+        return 0;
+    else
+        return n * twoToThePowerOf(b);
+}
+
+int rightShift(int n, int b) {
+    // assert: b >= 0
+    
+    if (b > 30)
+        return 0;
+    else if (n >= 0)
+        return n / twoToThePowerOf(b);
+    else
+        // works even if n == INT_MIN
+        return ((n + 1) + INT_MAX) / twoToThePowerOf(b) +
+        (INT_MAX / twoToThePowerOf(b) + 1);
+}
+
+
+void init (){
```

Comment:
So... your code starts here, right? I know... I told you to just copy the library to a new file, e.g. list.c, but now I realize that is hard to identify _you_ code. My fault. Next time, just add your code to selfie.c please